flixel/util/FlxDirection.hx

```diff
@@ -0,0 +1,50 @@
+package flixel.util;
+
+@:enum abstract FlxDirection(Int) from Int to Int
+{
+	var LEFT    = 0x0001;
+	var RIGHT   = 0x0010;
+	var UP      = 0x0100;
+	var DOWN    = 0x1000;
+	var NONE    = 0x0000;
+	var CEILING = 0x0100;// UP;
+	var FLOOR   = 0x1000;// DOWN;
+	var WALL    = 0x0011;// LEFT | RIGHT;
+	var ANY     = 0x1111;// LEFT | RIGHT | UP | DOWN;
+	
+	public inline function has(dir:FlxDirection):Bool return this & dir == dir;
+	
+	@:op(A & B)
+	inline function and(dir:FlxDirection):FlxDirection return this & dir;
```

Comment:
Not sure these `@:op` functions need bodies? If you omit it, that should function as "forwarding the operator".

Any particular reason these have `FlxDirection` as arguments, and the comparison operators `Int`?

Reply:
[this is why](https://try.haxe.org/#8d1A02B4) I'm using ints for some. Luckily [this works](https://try.haxe.org/#d60d4cad), I didn't know that existed.